### PR TITLE
Support Dashboard Metrics Storage Key

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve? 
close https://github.com/pingcap/tidb-operator/issues/2331

This request sync prometheus and grafana information into pd etcd with tidb nightly verison. I leave alertmanager into ucp. Here is the demo:

image

image

### Does this PR introduce a user-facing change?:

```release-note
Support `Dashboard` metrics ability for `TidbCluster` when `TidbMonitor` deployed 
```

### CODE REVIEW

File: pkg/manager/member/tidbcluster_status_manager.go [hunk 1]
```diff
@@ -14,37 +14,180 @@
...
+		err = cleanGrafanaKey(pdEtcdClient)
+		if err != nil {
+			klog.Error(err.Error())
+			return err
 		}
 	}
 	return nil
 }
 
+func putGrafanaKey(etcdClient pdapi.PDEtcdClient, value string) error {
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidbcluster_status_manager.go:150]: What's the benefit of wrapping up the etcdClient functions here?
The functions are just with different names only.

File: pkg/manager/member/tidbcluster_status_manager.go [hunk 2]
```diff
@@ -14,37 +14,180 @@
...
 func (tcsm *TidbClusterStatusManager) Sync(tc *v1alpha1.TidbCluster) error {
+	return tcsm.syncTidbMonitorRefAndKey(tc)
+}
+
+func (tcsm *TidbClusterStatusManager) syncTidbMonitorRefAndKey(tc *v1alpha1.TidbCluster) error {
+	tm, err := tcsm.syncTidbMonitorRef(tc)
+	if err != nil {
+		return err
+	}
+	return tcsm.syncDashboardMetricStorage(tc, tm)
```
Comment (DanielZhangQD) [inline pkg/manager/member/tidbcluster_status_manager.go:57]: Status manager should focus on the status update, can we move the etcd key operation to a separate function out of the status manager?

File: pkg/manager/member/tidbcluster_status_manager.go (hunk 1 above)
Comment (Yisaer) [reply pkg/manager/member/tidbcluster_status_manager.go:150]: Just want to make things clear and isolated.

File: pkg/manager/member/tidbcluster_status_manager.go (hunk 2 above)
Comment (Yisaer) [reply pkg/manager/member/tidbcluster_status_manager.go:57]: Currently, i think it is not necessary as it seems we would only solve the `MonitorRef` and `AutoSclaerRef` (in future) here. If we movet the etcd operation out of statusManager, we will try to fetch `TidbMonitor` again in its scope.

File: pkg/manager/member/tidbcluster_status_manager.go (hunk 1 above)
Comment (Yisaer) [reply pkg/manager/member/tidbcluster_status_manager.go]: updated.

File: pkg/pdapi/pdetcd.go [hunk 3]
```diff
@@ -19,11 +19,13 @@ import (
 	"time"
 
 	etcdclientv3 "github.com/coreos/etcd/clientv3"
+	etcdclientv3util "github.com/coreos/etcd/clientv3/clientv3util"
 )
 
-type PDEtcdApi interface {
+type PDEtcdClient interface {
+	// PutKey would put key to the target pd etcd cluster
```
Comment (cofyc) [inline pkg/pdapi/pdetcd.go:26]: why use `would` here? At the end of the function, the put/delete operation may not finish?

File: pkg/pdapi/pdetcd.go [hunk 4]
```diff
@@ -19,11 +19,13 @@ import (
...
 	PutKey(key, value string) error
-
+	// DeleteKey would delete key from the target pd etcd cluster
```
Comment (cofyc) [inline pkg/pdapi/pdetcd.go:28]: ditto

File: pkg/pdapi/pdetcd.go (hunk 3 above)
Comment (Yisaer) [reply pkg/pdapi/pdetcd.go:26]: I think the comment is misleading, I should use `will` here.

File: pkg/pdapi/pdapi.go [hunk 5]
```diff
@@ -72,6 +75,32 @@ func GetTLSConfig(kubeCli kubernetes.Interface, namespace Namespace, tcName stri
 	return crypto.LoadTlsConfigFromSecret(secret, caCert)
 }
 
+func (pdc *defaultPDControl) GetPDEtcdClient(namespace Namespace, tcName string, tlsEnabled bool) (PDEtcdClient, error) {
+	pdc.mutex.Lock()
```
Comment (DanielZhangQD) [inline pkg/pdapi/pdapi.go:79]: Suggest define separate mutex here, it's an independent client with existing PDClient.

Comment (Yisaer) [reply pkg/pdapi/pdapi.go]: updated.

**Pull request merged**